lib/MongoDbTransactionStore.ts

```diff
@@ -0,0 +1,235 @@
+import Transaction from './Transaction';
+import { Collection, MongoClient, Db } from 'mongodb';
+import { TransactionStore } from './TransactionStore';
+
+interface IUnresolvableTransaction extends Transaction {
+  firstFetchTime: number;
+  retryAttempts: number;
+  nextRetryTime: number;
+}
+
+/**
+ * Implementation of TransactionStore that stores the transaction data in a MongoDB database.
+ */
+export default class MongoDbTransactionStore implements TransactionStore {
+  /** Default database name used if not specified in constructor. */
+  public static readonly defaultDatabaseName: string = 'sidetree';
```

Comment:
I am worried that this could lead to subtle race condition bugs: In MongoDb, when a non-existent database (here, sidetree) is referenced the database is auto-created. What happens when TransactionStore and OperationStore reference and thereby set off database creation around the same time? If you can find documentation that Mongo correctly takes locks and handles this situation then we should be fine.

Reply:
Since we await initialization of `TransactionStore` and `OperationStore` we should be fine. let me know if you disagree.